src/components/flat-table/flat-table.component.tsx

```diff
@@ -203,17 +203,37 @@ export const FlatTable = ({
   };
 
   useLayoutEffect(() => {
-    const focusableElements = tableRef.current?.querySelectorAll(
-      FOCUSABLE_ROW_AND_CELL_QUERY
-    );
-
-    // if no other menu item is selected, we need to make the first row a tab stop
-    if (focusableElements && !selectedId) {
-      const focusableArray = Array.from(focusableElements).filter(
-        (el) => el.getAttribute("tabindex") !== null
+    const findSelectedId = () => {
+      const focusableElements = tableRef.current?.querySelectorAll(
+        FOCUSABLE_ROW_AND_CELL_QUERY
       );
-      setSelectedId(focusableArray[0]?.getAttribute("id") || "");
+
+      // if no other menu item is selected, we need to make the first row a tab stop
+      if (focusableElements && !selectedId) {
+        const selected = Array.from(focusableElements).find(
+          (el) => el.getAttribute("tabindex") !== null
```

Comment:
**comment** (non-blocking): since we're using `find` to only get the first such element (I presume there's only supposed to be at most one at a time), would it not be more efficient to use `querySelector` instead of `querySelectorAll`?